src/Arcus.Messaging.Abstractions/MessageHandling/MessageRouter.cs

```diff
@@ -477,66 +476,40 @@ protected virtual bool TryDeserializeToMessageFormat(string message, Type messag
 
             Logger.LogTrace("Try to JSON deserialize incoming message to message type '{MessageType}'...", messageType.Name);
 
-            var success = true;
-            JsonSerializer jsonSerializer = CreateJsonSerializer();
-            EventHandler<ErrorEventArgs> eventHandler = (sender, args) =>
-            {
-                success = false;
-                Logger.LogTrace(args.ErrorContext.Error, "Incoming message failed to be JSON deserialized to message type '{MessageType}' at {Path}", messageType.Name, args.ErrorContext.Path);
-                args.ErrorContext.Handled = true;
-            };
-            jsonSerializer.Error += eventHandler;
-            
+            JsonSerializerOptions options = CreateJsonSerializerOptions();
```

Comment:
Do we want to create a new instance of this class for every message that we process ?
I believe we can declare a static readonly member that represents the JsonSerializerOptions that we'd use, and by doing so we apply some kind of fly-weight :)